flaml/automl.py

```diff
@@ -624,6 +655,13 @@ def _prepare_data(self,
                 f"requires input data with at least {n_splits*2} examples.")
             self._state.kf = RepeatedStratifiedKFold(
                 n_splits=n_splits, n_repeats=1, random_state=RANDOM_SEED)
+        elif self._split_type == "time":
+            logger.info("Using TimeSeriesSplit")
+            if self._state.task == 'forecast':
+                self._state.kf = TimeSeriesSplit(
+                    n_splits=n_splits, test_size=self._state.fit_kwargs.get('period'))
+            else:
+                self._state.kf = TimeSeriesSplit(n_splits=n_splits)

```

Comment:
Is there a situation when time split is used but not for forecasting?

Reply:
yes, we're letting it be used for regression and classification tasks as well